python/ray/tests/kuberay/test_autoscaling_e2e.py

```diff
@@ -0,0 +1,243 @@
+        return raycluster_cr_file.name
+
+    def _get_ray_cr_config(
+        self, min_replicas=0, max_replicas=300, replicas=0
+    ) -> Dict[str, Any]:
+        """Get Ray CR config yaml, with configurable replica fields for the single
+        workerGroup."""
+        config = yaml.safe_load(open(self._get_ray_cr_config_file()).read())
+        config["spec"]["workerGroupSpecs"][0]["replicas"] = replicas
+        config["spec"]["workerGroupSpecs"][0]["minReplicas"] = min_replicas
+        config["spec"]["workerGroupSpecs"][0]["maxReplicas"] = max_replicas
+
+        return config
+
+    def _apply_ray_cr(self, min_replicas=0, max_replicas=300, replicas=0) -> None:
+        """Apply Ray CR config yaml, with configurable replica fields for the single
+        workerGroup."""
+        with tempfile.NamedTemporaryFile("w") as config_file:
+            cr_config = self._get_ray_cr_config(
+                min_replicas=min_replicas, max_replicas=max_replicas, replicas=replicas
+            )
+            yaml.dump(cr_config, config_file)
+            config_file.flush()
+            subprocess.check_call(["kubectl", "apply", "-f", config_file.name])
+
+    def testAutoscaling(self):
+        """Test the following behaviors:
+
+        1. Spinning up a Ray cluster
+        2. Scaling up a Ray worker via autoscaler.sdk.request_resources()
+        3. Scaling up by updating the CRD's minReplicas
+        4. Scaling down by removing the resource request and reducing maxReplicas
+
+        Items 1. and 2. protect the example in the documentation.
+        Items 3. and 4. protect the autoscaler's ability to respond to Ray CR update.
+
+        Resources requested by this test are safely within the bounds of an m5.xlarge
+        instance.
+        - One Ray head pod
+            - Autoscaler: .5 CPU, .5 Gi memory
+            - Ray node: .5 CPU, .5 Gi memeory
+        - Two Worker pods
+            - Ray node: .5 CPU, .5 Gi memory
+        Total: 2 CPU, 2 Gi memory.
+
+        Including operator and system pods, the total CPU requested is just over 3.
+        """
+        # Cluster-creation
+        logger.info("Creating a RayCluster with no worker pods.")
+        self._apply_ray_cr(min_replicas=0, replicas=0)
+
+        logger.info("Confirming presence of head.")
+        wait_for_pods(goal_num_pods=1, namespace="default")
+        head_pod = get_pod(
+            pod_name_filter="raycluster-complete-head", namespace="default"
+        )
+
+        logger.info("Waiting for head pod to start Running.")
+        wait_for_pod_to_start(head_pod, namespace="default")
+        logger.info("Confirming Ray is up on the head pod.")
+        wait_for_ray_health(head_pod, namespace="default")
+
+        # Scale-up
+        logger.info("Scaling up to one worker via Ray resource request.")
+        scale_script = (
+            "import ray;"
+            'ray.init("auto");'
+            "ray.autoscaler.sdk.request_resources(num_cpus=2)"
```

Comment:
We expect 1 worker (0.5 cpu) created after `"ray.autoscaler.sdk.request_resources(num_cpus=2)"`?   0.5 worker + 0.5 head = 1 cpu. 

```
- One Ray head pod
          - Autoscaler: .5 CPU, .5 Gi memory
          - Ray node: .5 CPU, .5 Gi memeory
- Two Worker pods
           - Ray node: .5 CPU, .5 Gi memory
```

Reply:
I will add some clarifying comments: the request is .5, the limit is 1, and the resource annotation (--num-cpus) in Ray start is 1. 
So from Ray's perspective the head and each worker have 1 CPU.

Reply:
Added more to the test docstring, added a comment in the code.